functions/_pure_prompt_nixdevshell_type.fish

```diff
@@ -0,0 +1,18 @@
+function _pure_prompt_nixdevshell_type \
+    --description "Indicates the nix shell type"
+
+    # Ported starship's implementation to fish.
+    # https://github.com/starship/starship/blob/9ab8b84ea6cd40f83a34836111c6bda2203dccfe/src/modules/nix_shell.rs#L15-L40
+
+    if set --query IN_NIX_SHELL
+        echo $IN_NIX_SHELL
+    else
+        for path in $PATH
+            # glob pattern seems to be broken in fish?
+            if string match --regex --quiet -- /nix/store $path
+                echo unknown
+                break
+            end
+        end
```

Comment:
If you want to detect `/nix/store`, you can directly match the `$PATH` variable

```suggestion
        string match --regex --quiet -- "/nix/store" $PATH
```

Reply:
The initial plan was to not use regex for performance reasons, but I did not manage to get glob patterns a working.